File: frontend/src/types/github.ts
export interface RepoScan {
  id: string;
  repo_id: string;
  health_score?: number;
  resume_score?: number;
  portfolio_tier?: string;
  key_technologies?: string[];
  architecture_summary?: string;
  resume_bullets?: string[];
  recommendation_reason?: string;
  production_readiness?: {
    has_tests?: boolean;
    has_docker?: boolean;
    has_ci_cd?: boolean;
    code_quality_rating?: string;
  };
  has_gitignore?: boolean;
  has_env_file?: boolean;
  leaked_secrets?: any[];
  ai_issues?: any[];
  scanned_at: string;
}

export interface GithubRepo {
  id: string;
  github_id: number;
  name: string;
  full_name: string;
  description?: string;
  language?: string;
  topics?: string[];
  has_readme: boolean;
  readme_content?: string;
  is_private: boolean;
  stars: number;
  forks_count?: number;
  open_issues_count?: number;
  size_kb?: number;
  default_branch?: string;
  license_name?: string;
  html_url?: string;
  last_pushed_at?: string;
  synced_at: string;
  latest_scan?: RepoScan;
}

export interface ReadmeGenerateResponse {
  readme_markdown: string;
  suggestion_id: string;
}

export interface ReadmePushResponse {
  committed: boolean;
  sha: string;
}

export interface RemediateResponse {
  repo_full_name: string;
  remediated: boolean;
  action_taken: string;
  commit_sha?: string;
  message: string;
}

File: backend/models/github.py
from sqlalchemy import Column, String, Integer, BigInteger, Boolean, DateTime, ForeignKey, text, ARRAY
from sqlalchemy.dialects.postgresql import UUID, JSONB
from sqlalchemy.sql import func
from database import Base

class GithubRepo(Base):
    __tablename__ = 'github_repos'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"))
    github_id = Column(BigInteger, unique=True, nullable=False)
    name = Column(String, nullable=False)
    full_name = Column(String, nullable=False)
    description = Column(String)
    language = Column(String)
    topics = Column(ARRAY(String))
    has_readme = Column(Boolean, server_default=text("false"))
    readme_content = Column(String)
    is_private = Column(Boolean, server_default=text("false"))
    stars = Column(Integer, server_default=text("0"))
    forks_count = Column(Integer, server_default=text("0"))
    open_issues_count = Column(Integer, server_default=text("0"))
    size_kb = Column(Integer, server_default=text("0"))
    default_branch = Column(String, default="main")
    license_name = Column(String, nullable=True)
    html_url = Column(String, nullable=True)
    last_pushed_at = Column(DateTime(timezone=True))
    synced_at = Column(DateTime(timezone=True), server_default=func.now())

class RepoScan(Base):
    __tablename__ = 'repo_scans'

    id = Column(UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()"))
    repo_id = Column(UUID(as_uuid=True), ForeignKey('github_repos.id', ondelete='CASCADE'))
    health_score = Column(Integer)
    resume_score = Column(Integer)
    portfolio_tier = Column(String)
    key_technologies = Column(ARRAY(String))
    architecture_summary = Column(String)
    resume_bullets = Column(JSONB)
    recommendation_reason = Column(String)
    production_readiness = Column(JSONB)
    has_gitignore = Column(Boolean)
    has_env_file = Column(Boolean)
    leaked_secrets = Column(JSONB)
    ai_issues = Column(JSONB)
    scanned_at = Column(DateTime(timezone=True), server_default=func.now())
